pkg/console/service/kubevirt-proxy.go

```diff
@@ -1,29 +1,253 @@
 package service
 
 import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
 	"github.com/emicklei/go-restful/v3"
+	authnv1 "k8s.io/api/authentication/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	kubevirt "kubevirt.io/api/core"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"github.com/kubevirt/vm-console-proxy/api/v1alpha1"
+)
+
+const (
+	AppKubernetesNameLabel      = "app.kubernetes.io/name"
+	AppKubernetesPartOfLabel    = "app.kubernetes.io/part-of"
+	AppKubernetesVersionLabel   = "app.kubernetes.io/version"
+	AppKubernetesManagedByLabel = "app.kubernetes.io/managed-by"
+	AppKubernetesComponentLabel = "app.kubernetes.io/component"
 )
 
-func NewKubevirtProxy() Service {
-	return &kubevirtProxy{}
+func NewKubevirtProxy(kubevirtClient kubecli.KubevirtClient) Service {
+	return &kubevirtProxy{
+		kubevirtClient: kubevirtClient,
+	}
 }
 
 type kubevirtProxy struct {
+	kubevirtClient kubecli.KubevirtClient
 }
 
 var _ Service = &kubevirtProxy{}
 
 func (k *kubevirtProxy) TokenHandler(request *restful.Request, response *restful.Response) {
-	//TODO: implement
-	panic("not implemented")
+	err := ValidateVncRbac(request, k.kubevirtClient)
+	if err != nil {
+		_ = response.WriteError(http.StatusUnauthorized, err)
+		return
+	}
+
+	namespace := request.PathParameter("namespace")
```

Comment:
Just out of curiosity, is there or will be there some parameters validation? e.g. namespace is empty, and we don't want to pass it to other functions.

Reply:
I've added validation.